docs/src/pages/components/buttons/UseButton.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import Stack from '@material-ui/core/Stack';
+import {
+  ButtonUnstyledProps,
+  useButton,
+} from '@material-ui/unstyled/ButtonUnstyled';
+import { styled } from '@material-ui/system';
+
+const StyledButtonRoot = styled('span')(`
```

Comment:
I don't think that it makes sense to use span here. It's not how most developers will use it. I think that we should encourage custom buttons to be created with a `<button>`:
```suggestion
const StyledButtonRoot = styled('button')(`
```

If we want to showcase that it also works with other elements, then I would advocate for a `<CustomButton component="span" />` to make it explicit in the inline demo preview.

Reply:
Oh I see we have something close in https://github.com/mui-org/material-ui/pull/27600/files#r683489011. Could we comment on the .tsx version only of the demos? 👼

Reply:
I'm still working on the docs. Perhaps it'll make more sense when it's finished. I intend to have a first demo using a `button`, and then, under a separate subsection, another one showing that you can use a different, non-interactive element and everything would work the same way.